worker-workflow/src/main/java/com/github/cafdataprocessing/workflow/ArgumentsManager.java

```diff
@@ -208,12 +274,126 @@ else if (type.equals("cd")){
+            return true;
+        }  else {
+            LOG.debug(String.format("NOT forcing cache refresh for: %s because the last update time: %s is NOT greater "
+                + "than the last access time: %s", key, settingsServiceLastUpdateTimeMillisOpt.get(),
+                settingsServiceLastAccessTimeMillis));
+            return false;
+        }
+    }
+
+    private ResolvedSetting forceCacheRefresh(final SettingsServiceApiCall settingsServiceApiCall) throws ApiException
+    {
+        try {
+            okHttpClient.interceptors().add(forceCacheRefreshInterceptor);
+            return settingsServiceApiCall.call();
+        } finally {
+            okHttpClient.interceptors().removeIf(ForceCacheRefreshInterceptor.class::isInstance);
+        }
+    }
+
+    @FunctionalInterface
+    private interface SettingsServiceApiCall
+    {
+        public ResolvedSetting call() throws ApiException;
+    }
+
+    private static final class SettingsServiceLastAccessTimeMapKey
+    {
+        private final String key;
+
+        private SettingsServiceLastAccessTimeMapKey(final String key)
+        {
+            this.key = key;
+        }
+
+        public static SettingsServiceLastAccessTimeMapKey from(final URI uri)
+        {
+            final String key = String.format("%s?%s", uri.getPath(), uri.getQuery());
+            return new SettingsServiceLastAccessTimeMapKey(key);
+        }
+
+        public static SettingsServiceLastAccessTimeMapKey from(
+            final String settingName,
+            final List<String> scopes,
+            final List<String> priorities)
+        {
+            final String scopesUrlQueryParam = (scopes != null && !scopes.isEmpty())
+                ? "scopes=" + String.join(",", scopes)
+                : null;
+            final String prioritiesUrlQueryParam = (priorities != null && !priorities.isEmpty())
+                ? "priorities=" + String.join(",", priorities)
+                : null;
+            final String urlQueryParams = Joiner.on("&").skipNulls().join(
+                scopesUrlQueryParam,
+                prioritiesUrlQueryParam
+            );
+            final String key = urlQueryParams != null
+                ? String.format("/settings/%s/resolved?%s", settingName, urlQueryParams)
+                : String.format("/settings/%s/resolved", settingName);
+            return new SettingsServiceLastAccessTimeMapKey(key);
+        }
+
+        @Override
+        public int hashCode()
+        {
+            int hash = 5;
+            hash = 97 * hash + Objects.hashCode(this.key);
+            return hash;
+        }
+
+        @Override
+        public boolean equals(Object obj)
+        {
+            if (this == obj) {
+                return true;
+            }
+            if (obj == null) {
+                return false;
+            }
+            if (getClass() != obj.getClass()) {
```

Comment:
Since `SettingsServiceLastAccessTimeMapKey` is `final` you can do this instead:
```suggestion
            if (!(obj instanceof SettingsServiceLastAccessTimeMapKey)) {
```